crates/rrg/src/fs.rs

```diff
@@ -203,7 +189,21 @@ impl std::iter::Iterator for ListDir {
     type Item = std::io::Result<Entry>;
```

Comment:
Is this needed?

Reply:
I am not sure what you mean by "needed". `Item` [is an associated type][1] on the `Iterator` trait.

[1]: https://doc.rust-lang.org/std/iter/trait.Iterator.html#required-associated-types

Reply:
Oh I see, thank you, I thought it is a [type alias](https://doc.rust-lang.org/reference/items/type-aliases.html) which is then not used.

Reply:
Ah, okay. Just so you know then: if you see `type ... = ...;` declaration in trait implementation block (`impl ... for ... { ... }`), it _has to_ be an associated type, traditional aliases are not allowed there.